### PR TITLE
Ads: ES6ify earnings form

### PR DESCRIPTION
**Pull request by tyxla**
This PR ES6ifies the ads earnings form and removes all ESLint warnings from it. This will help us do any further modifications without much changes while working on the upcoming ads modifications - https://github.com/Automattic/wp-calypso/milestone/169

There should be no functional, visual or behavioral changes.

To test:
* Checkout this branch
* Go to `/ads/earnings/$site` where `$site` is a Jetpack site with a Premium or Professional plan.
* Verify the earnings are properly listed and there are no visual/behavioral changes.
* Play with the info toggles and verify they work like they did before.
* Switch tabs, change some earnings settings, go back and verify everything works properly.
* Test the above with a WordPress.com site and verify there are no regressions.

### CODE REVIEW

File: client/my-sites/ads/form-earnings.jsx [hunk 1]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
-		var period,
-			rows = [],
+	earningsTable( earnings, header_text, type ) {
+		const { numberFormat, translate } = this.props;
+		const rows = [],
 			infoIcon = this.getInfoToggle( type ) ? 'info' : 'info-outline',
 			classes = classNames( 'earnings_history', {
 				'is-showing-info': this.getInfoToggle( type )
 			} );
+		let period;
```
Comment (sirreal) [inline client/my-sites/ads/form-earnings.jsx]: We don't need `period` in this scope, I believe we should be able to declare it as a `const` in the `for` loop:

```js
for ( const period in earnings ) {
```

File: client/my-sites/ads/form-earnings.jsx [hunk 2]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
 		event.preventDefault();
-		this.setState( { showEarningsNotice: ! this.state.showEarningsNotice } );
-	},
+		this.setState( {
+			showEarningsNotice: ! this.state.showEarningsNotice
+		} );
+	};
 
-	toggleInfo: function( type, event ) {
+	handleInfoToggle = ( type ) => ( event ) => {
```
Comment (sirreal) [inline client/my-sites/ads/form-earnings.jsx]: 👍  coherent names and currying FTW

File: client/my-sites/ads/form-earnings.jsx [hunk 3]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
-				this.setState( { showSponsoredInfo: ! this.state.showSponsoredInfo } );
+				this.setState( {
+					showSponsoredInfo: ! this.state.showSponsoredInfo,
+				} );
 				break;
 			case 'adjustment':
-				this.setState( { showAdjustmentInfo: ! this.state.showAdjustmentInfo } );
+				this.setState( {
+					showAdjustmentInfo: ! this.state.showAdjustmentInfo,
+				} );
```
Comment (sirreal) [inline client/my-sites/ads/form-earnings.jsx]: Did you consider folding these into one `case`? Your call as to whether it's improves readability or may reduce likelyhood of bugs.

```js
case 'adjustment':
case 'sponsored':
case 'wordads':
	this.setState( s => ( {
		[type]: ! s[type],
	} ) );
	break;
```

File: client/my-sites/ads/form-earnings.jsx [hunk 4]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
 		}, 0 );
-	},
+	}
 
-	toggleEarningsNotice: function( event ) {
+	updateSettings = () => {
+		this.setState( this.getSettingsFromStore() );
+	};
+
+	handleEarningsNoticeToggle = ( event ) => {
```
Comment (sirreal) [inline client/my-sites/ads/form-earnings.jsx]: Maybe I'm going overboard, but this is pretty much identical to `handleInfoToggle`. One of them could easily be removed.

File: client/my-sites/ads/form-earnings.jsx [hunk 5]
```diff
@@ -1,62 +1,52 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	notices = require( 'notices' ),
-	classNames = require( 'classnames' ),
-	debug = require( 'debug' )( 'calypso:my-sites:ads-earnings' );
+import React, { Component } from 'react';
+import notices from 'notices';
+import classNames from 'classnames';
```
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: I rarely alphabetize those, because we usually keep the `React` import at the top.

File: client/my-sites/ads/form-earnings.jsx [hunk 6]
```diff
@@ -1,62 +1,52 @@
...
-module.exports = React.createClass( {
+class AdsFormEarnings extends Component {
+	state = this.getSettingsFromStore();
 
-	displayName: 'AdsFormEarnings',
-
-	componentWillMount: function() {
-		debug( 'Mounting ' + this.constructor.displayName + ' React component.' );
+	componentWillMount() {
 		EarningsStore.on( 'change', this.updateSettings );
```
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: :+1: 0e5bb97bbc8477172fecf6ad8c73c74e1481c3ed

File: client/my-sites/ads/form-earnings.jsx [hunk 7]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
 			showEarningsNotice: false,
 			showWordadsInfo: false,
 			showSponsoredInfo: false,
-			showAdjustmentInfo: false
+			showAdjustmentInfo: false,
 		} );
```
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: I'll pass here.

File: client/my-sites/ads/form-earnings.jsx [hunk 8]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
-			debug( 'initial fetch not necessary' );
 			return;
 		}
 
 		// defer fetch requests to avoid dispatcher conflicts
-		setTimeout( function() {
-			WordadsActions.fetchEarnings( site )
+		setTimeout( () => {
+			WordadsActions.fetchEarnings( site );
 		}, 0 );
```
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: :+1: 2e33e48ef70965ea89dedeb27a7671206e3c07be

File: client/my-sites/ads/form-earnings.jsx (hunk 4 above)
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: Let's leave that for another PR.

File: client/my-sites/ads/form-earnings.jsx (hunk 3 above)
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: Let's leave that for another PR. It's going well out of this PR's purpose.

File: client/my-sites/ads/form-earnings.jsx [hunk 9]
```diff
@@ -81,198 +71,229 @@ module.exports = React.createClass( {
...
+	}
 
-	payoutNotice: function() {
-		var owed = this.state.earnings && this.state.earnings.total_amount_owed ? this.state.earnings.total_amount_owed : '0.00',
-			notice = this.translate(
-				'Outstanding amount of $%(amountOwed)s does not exceed the minimum $100 needed to make the payment. Payment will be made as soon as the total outstanding amount has reached $100.',
+	payoutNotice() {
+		const { translate } = this.props;
+		const owed = this.state.earnings && this.state.earnings.total_amount_owed ? this.state.earnings.total_amount_owed : '0.00',
+			notice = translate(
```
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: I'm also inclined to do that, but since we don't have a clear standard for which one is better, I'll leave them as they are.

File: client/my-sites/ads/form-earnings.jsx (hunk 1 above)
Comment (tyxla) [inline client/my-sites/ads/form-earnings.jsx]: :+1: ddaba4a2ba703140855051732a1fafed746270cc

**Comment by tyxla** [comment]
@sirreal thanks for the feedback! I've addressed what I felt should be addressed in this PR. Feel free to have another 👀
